cmd/purge/networks/networks.go

```diff
@@ -56,6 +61,32 @@ pvsadm purge --help for information
 		if !opt.DryRun && len(networks) != 0 {
 			if opt.NoPrompt || utils.AskYesOrNo(deletePromptMessage) {
 				for _, network := range networks {
+					if deleteInstances || deletePorts {
+						ports, err := pvmclient.NetworkClient.GetAllPorts(*network.NetworkID)
+						if err != nil {
+							return fmt.Errorf("failed to get the list of ports: %v", err)
+						}
+
+						// Clean up instances and ports associated with the network instance
+						for _, port := range ports.Ports {
+							pvminstance := port.PvmInstance
```

Comment:
there are chances that this field may be a nil, if nil then good to skip deleting the instance and directly delete the port

Reply:
the case where you have the port left behind used for the HA in the capi